src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1164,13 +1171,60 @@ private boolean confirmClose(LibraryTab libraryTab) {
         return response.isEmpty() || !response.get().equals(cancel);
     }
 
+    /**
+     * Ask if the user really wants to remove any empty entries
+     */
+    private Boolean confirmEmptyEntry(LibraryTab libraryTab, BibDatabaseContext context) {
+        String filename = libraryTab.getBibDatabaseContext()
+                                    .getDatabasePath()
+                                    .map(Path::toAbsolutePath)
+                                    .map(Path::toString)
+                                    .orElse(Localization.lang("untitled"));
+
+        ButtonType deleteEmptyEntries = new ButtonType(Localization.lang("Delete empty entries"), ButtonBar.ButtonData.YES);
+        ButtonType keepEmptyEntries = new ButtonType(Localization.lang("Keep empty entries"), ButtonBar.ButtonData.NO);
+        ButtonType cancel = new ButtonType(Localization.lang("Return to JabRef"), ButtonBar.ButtonData.CANCEL_CLOSE);
+
+        Optional<ButtonType> response = dialogService.showCustomButtonDialogAndWait(Alert.AlertType.CONFIRMATION,
+                Localization.lang("Empty entries"),
+                Localization.lang("Library '%0' has empty entries. Do you want to delete them?", filename),
+                deleteEmptyEntries, keepEmptyEntries, cancel);
+        if (response.isPresent() && response.get().equals(deleteEmptyEntries)) {
+            // The user wants to delete.
+            try {
+                for (BibEntry currentEntry : new ArrayList<BibEntry>(context.getEntries())) {
```

Comment:
Why do you create a new array list here?

Reply:
If I don't create a new array list, removing the empty entry from the original array list will lead to out of index error.

Reply:
Are there any other approaches you would recommend? @Siedlerchr

Reply:
Ah yes, now I see it. It's because you can't remove an entry from a collection over that you are still iterating. So your solutions is fine.
Please add one more test  where you check and verify that the context.getEntries() is indeed entry 
e.e.g assertEquals(Collections.emptyList(), context.getEntries())

Reply:
Yes, exactly.
Okay, will do.